MarsRover.Tests/ControllerShould.cs

```diff
@@ -6,67 +6,84 @@ namespace MarsRover.Tests
+        }
         [Fact]
         public void Run_ReturnsMapOutput_GivenRoverCommandsThatResultInNoObstacle()
         {
-            var mockInput = new Mock<IInput>();
-            mockInput.SetupSequence(_ => _.ReadLine())
+            _mockInput.SetupSequence(_ => _.ReadLine())
                 .Returns("TestFiles/validFile1.txt")
                 .Returns("N")
                 .Returns("1.0")
                 .Returns("2,0")
                 .Returns("lfrlb")
                 .Returns("q");
-            var output = new StubOutput();
-            var fileMapInput = new FileMapInput();
             var expectedString = "🟫⬜️⏪⬜️\n⬜️⬜️⬜️⬜️\n⬜️⬜️⬜️⬜️";
-            var controller = new Controller(mockInput.Object, output, fileMapInput);
-
-            controller.Run();
+            
+            _controller.Run();
 
-            output.GetLastMapOutput().Should().BeEquivalentTo(expectedString);
+            _output.GetLastMapOutput().Should().BeEquivalentTo(expectedString);
         }
 
         [Fact]
         public void Run_ReturnsRoverReport_GivenRoverCommandsThatResultInAnObstacle()
         {
-            var mockInput = new Mock<IInput>();
-            mockInput.SetupSequence(_ => _.ReadLine())
+            _mockInput.SetupSequence(_ => _.ReadLine())
                 .Returns("TestFiles/validFile1.txt")
                 .Returns("N")
                 .Returns("1.0")
                 .Returns("1,0")
                 .Returns("lfr")
                 .Returns("q");
-            var output = new StubOutput();
-            var fileMapInput = new FileMapInput();
             var expectedString = "Rover can't move. Obstacle at 0,0.";
-            var controller = new Controller(mockInput.Object, output, fileMapInput);
 
-            controller.Run();
+            _controller.Run();
 
-            output.OutputList.Should().Contain(expectedString);
+            _output.OutputList.Should().Contain(expectedString);
         }
 
         [Fact]
         public void Run_ReturnsRoverReport_GivenRoverCommandsThatResultInAnObstacleAnd2ConsecutiveCommandsSame()
         {
-            var mockInput = new Mock<IInput>();
-            mockInput.SetupSequence(_ => _.ReadLine())
+            _mockInput.SetupSequence(_ => _.ReadLine())
                 .Returns("TestFiles/validFile1.txt")
                 .Returns("N")
                 .Returns("1.0")
                 .Returns("2,0")
                 .Returns("lff")
                 .Returns("q");
-            var output = new StubOutput();
-            var fileMapInput = new FileMapInput();
             var expectedString = "Rover can't move. Obstacle at 0,0.";
-            var controller = new Controller(mockInput.Object, output, fileMapInput);
+            
+            _controller.Run();
+
+            _output.OutputList.Should().Contain(expectedString);
+        }
+
+        [Fact]
+        public void Run_ReturnsEndMessage_Given2SetsOfRoverCommands()

```

Comment:
The test should probably care about the final state of the map/rover i.e. given this map + set of commands, where do we expect the rover to be at the end?

Reply:
Is it the name you're referring to? i.e Run_ReturnsEndMessage_GiveMapAnd@SetsOfRoverCommands
I didn't think I should check the Rover's final location - testing implementation not behaviour.

Reply:
@lindajohnstone testing the rover's final location here is not testing implementation details. Think of the commands you give as the input, and the rover's final location as the output. Given a set of commands, if your application is working correctly, the rover is expected to end up at an expected location of the given map. The end message is not a good indicator of your application working correctly - how do you know your run loop has executed correctly without checking that the rover is where it's supposed to be?